files/api/activity.rb

```diff
@@ -11,27 +11,48 @@ def initialize(pe_console, username: nil, password: nil, token: nil, ssl_verify:
       @pe_client = PeEventForwarding::PeHttp.new(pe_console, port: 4433, username: username, password: password, token: token, ssl_verify: ssl_verify)
     end
 
-    def get_events(service: nil, offset: nil, limit: nil, order: 'asc')
+    def get_events(service: nil, offset: 0, order: 'asc', api_window_size: nil)
       params = {
         service_id: service,
-        limit:      limit,
+        limit:      api_window_size,
         offset:     offset,
         order:      order,
       }
-      response = pe_client.pe_get_request('activity-api/v2/events', params)
+
+      api_window_size = api_window_size.to_i
+      response_items = []
+      response = ''
+      total_count = 0
+      loop do
+        response = pe_client.pe_get_request('activity-api/v2/events', params)
+        response_body = JSON.parse(response.body)
+        total_count = response_body['pagination']['total']
+        response_items = response_items << response_body['items']
```

Comment:
You don't need to do the `<<` concatenation and then assign. The `<<` is an assignment operator, so you can just do that on a line and move on with no need for the `=`.

Reply:
And could these `=` operators be aligned here as in the comment from the lines above? You can align the `<<` with the `=`.